backend/benches/benchmark.rs

```diff
@@ -0,0 +1,105 @@
+use chrono::{DateTime, Duration, TimeDelta, Utc};
+use criterion::{criterion_group, criterion_main, Criterion};
+
+use rand::Rng;
+use scheduling_backend::data_model::graph::DiscreteGraph;
+use scheduling_backend::scheduling::scheduler::{GlobalSchedulerAlgorithm, SchedulerAlgorithm};
+use scheduling_backend::scheduling::{
+    scheduler::NaiveSchedulerAlgorithm, task_for_scheduler::TaskForScheduler,
+};
+
+use protocol::{tasks::TaskId, time::Timespan};
+
+struct TaskFactory {
+    task_id: TaskId,
+}
+impl TaskFactory {
+    fn get_task_id(&mut self) -> TaskId {
+        let res = self.task_id;
+        self.task_id += 1;
+        res
+    }
+    pub fn new() -> Self {
+        TaskFactory { task_id: 0.into() }
+    }
+    pub fn make_tasks(&mut self, amount: usize, start: DateTime<Utc>) -> Vec<TaskForScheduler> {
+        let mut res = Vec::new();
+        for _ in 0..amount {
+            let mut rng = rand::thread_rng();
+            let timespan_start = Duration::seconds(0).num_seconds();
+            let timespan_end = Duration::hours(24).num_seconds();
+
+            let start_offset = Duration::seconds(rng.gen_range(timespan_start..timespan_end));
+            let end_offset =
+                Duration::seconds(rng.gen_range(start_offset.num_seconds()..timespan_end))
+                    + Duration::seconds(1);
+
+            let start_time = start + start_offset;
+            let end_time = start + end_offset;
+
+            let total_duration = (end_time - start_time).num_seconds();
+            let duration = rng.gen_range(1..=total_duration);
+
+            let max_effect = 10000.0;
+            let effect = rng.gen_range(1.0..=max_effect);
+
+            res.push(TaskForScheduler {
+                id: self.get_task_id(),
+                timespan: Timespan {
+                    start: start_time,
+                    end: end_time,
+                },
+                duration: duration.into(),
+                effect,
+            });
+        }
+        res
+    }
+}
+
+fn naive_scheduling_benchmark(c: &mut Criterion) {
```

Comment:
Byg funktionerne med hensigt på at teste ud fra to parametre, antal tasks, og antal time-slots.
F.eks. vil vi lave tests hvor der altid er 24 timeslots, men henholdsvis 100, 1000, 10000 tasks.
Og omvendt, hvor det altid er f.eks. 100 tasks, men hvor antallet af timeslots ændres fra 24, til 240, 2400 osv.